src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -615,20 +611,69 @@ def launch_fluent(
 
     del start_container
 
-    if start_watchdog is None and cleanup_on_exit:
+    if additional_arguments is None:
+        additional_arguments = ""
+    elif fluent_launch_mode == LaunchMode.PIM:
+        raise ValueError(
+            "'additional_arguments' option for 'launch_fluent' is currently not supported "
+            "when starting a remote Fluent PyPIM client."
+        )
+
+    if fluent_launch_mode == LaunchMode.PIM and start_watchdog:
+        raise ValueError(
+            "'start_watchdog' argument for 'launch_fluent' is currently not supported "
+            "when starting a remote Fluent PyPIM client."
+        )
+
+    if (
+        start_watchdog is None
+        and cleanup_on_exit
+        and (
+            fluent_launch_mode == LaunchMode.CONTAINER
+            or fluent_launch_mode == LaunchMode.STANDALONE
+        )
+    ):
         start_watchdog = True
 
-    if dry_run:
-        if not fluent_launch_mode == LaunchMode.CONTAINER:
+    if dry_run and fluent_launch_mode != LaunchMode.CONTAINER:
+        raise ValueError(
+            "'dry_run' argument for 'launch_fluent' currently is only "
+            "supported when starting containers."
+        )
+
+    argvals = locals()
+
+    if fluent_launch_mode != LaunchMode.STANDALONE:
+        arg_names = [
+            "env",
+            "cwd",
+            "topy",
+            "case_filepath",
+            "lightweight_mode",
+            "journal_filepath",
+            "case_data_filepath",
+        ]
+        invalid_arg_names = []
+        for arg_name in arg_names:
+            if argvals[arg_name] is not None:
+                invalid_arg_names.append(arg_name)
```

Comment:
```suggestion
        invalid_arg_names=filter(lambda arg_name:argvals[arg_name] is not None, arg_names)
```

(for example)

Reply:
I knew there had to be a better way to do that. Wasn't familiar with `filter()`, that is going to be useful, thanks Sean

Reply:
`invalid_arg_names = [arg_name for arg_name in arg_names if argvals[arg_name] is not None]`

if you prefer list-comprehension

Reply:
Using list comprehension is also better than what I suggested initially, thanks Mainak. 

Between filter and list comprehension, either one works fine I think, going to stick with filter for this one as it is an interesting built-in function :)

Reply:
Bear in mind that there is a performance difference. One of these is almost twice as quick a the other:
```
>>> xs = range(100000000)
>>> ys = list(filter(lambda x: x%2, xs))
>>> zs = [x for x in xs if x%2]
```
but it doesn't really matter here.